zulip_bots/zulip_bots/provision.py

```diff
@@ -1,67 +1,85 @@
 #!/usr/bin/env python
 
 from __future__ import absolute_import
-from __future__ import print_function
 
 import argparse
+import logging
 import os
 import sys
+import glob
 import pip
 
 def provision_bot(path_to_bot, force):
     # type: (str, bool) -> None
     req_path = os.path.join(path_to_bot, 'requirements.txt')
     install_path = os.path.join(path_to_bot, 'bot_dependencies')
+
     if os.path.isfile(req_path):
-        print('Installing dependencies...')
+        bot_name = os.path.basename(path_to_bot)
+        logging.info('Installing dependencies for {}...'.format(bot_name))
         if not os.path.isdir(install_path):
             os.makedirs(install_path)
+
         # pip install -r $BASEDIR/requirements.txt -t $BASEDIR/bot_dependencies --quiet
         rcode = pip.main(['install', '-r', req_path, '-t', install_path, '--quiet'])
-        if not rcode == 0:
-            print('Error. Check output of `pip install` above for details.')
+
+        if rcode != 0:
+            logging.error('Error. Check output of `pip install` above for details.')
             if not force:
-                print('Use --force to try running anyway.')
+                logging.error('Use --force to try running anyway.')
                 sys.exit(rcode)  # Use pip's exit code
-        else:
-            print('Installed.')
-        sys.path.insert(0, install_path)
-
-def dir_join(dir1, dir2):
-    # type: (str, str) -> str
-    return os.path.abspath(os.path.join(dir1, dir2))
 
-def run():
-    # type: () -> None
-    usage = '''
-        Installs dependencies of bots in api/bots directory. Add a
-        reuirements.txt file in a bot's folder before provisioning.
-
-        To provision all bots, use:
-        ./provision.py
+        logging.info('Installed dependencies successfully.')
+        sys.path.insert(0, install_path)
```

Comment:
Correct me if I am wrong, but since the default value of `force` is `False`, that means that if there are errors, then the `provision.py` will `sys.exit` *before* it adds the `intall_dir` to `sys.path`.

If `force` is `True` and if there are pip errors, then it will move on and add `install_dir` to `sys.path` anyway. Isn't that how it should be? Or am I missing something here and maybe there's a better way?